src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -151,6 +151,8 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
             // Check the item count.
             if (searchParams.Count != null)
             {
+                searchOptions.MaxItemCountSpecifiedByClient = true;
```

Comment:
Should this only be true if we use the value for Count?

Reply:
It should be true iff `_count` is specified in the search.

Reply:
Sorry I was unclear. Below this we ignore the _count if it is higher than the MaxItemCountPerSearch. Should this still be true if we ignore the _count?

Reply:
Oh I see what you mean. When this is true, we treat the MaxItemCount as a hard limit. When _count is higher than MaxItemCountPerSearch and we set it to MaxItemCountPerSearch, I think we should treat it as a hard limit.